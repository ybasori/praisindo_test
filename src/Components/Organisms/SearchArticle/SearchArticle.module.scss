.artilce-list {
  margin-top: 8px;
}
