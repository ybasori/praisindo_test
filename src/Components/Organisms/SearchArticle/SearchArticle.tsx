import React from "react";
import styles from "./SearchArticle.module.scss";
import SearchInput from "../../Atoms/SearchInput/SearchInput";
import ArticleList from "../../Molecules/ArticleList/ArticleList";

const SearchArticle = () => {
  return (
    <>
      <div className={styles["container"]}>
        <SearchInput />
        <div className={styles["artilce-list"]}>
          <ArticleList />
        </div>
      </div>
    </>
  );
};

export default SearchArticle;
