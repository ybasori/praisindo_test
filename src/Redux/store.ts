import { configureStore } from "@reduxjs/toolkit";
import nyTimesReducer from "./NyTimes/NyTimes.reducer";

export const store = configureStore({
  reducer: {
    nyTimes: nyTimesReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ serializableCheck: false }),
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
