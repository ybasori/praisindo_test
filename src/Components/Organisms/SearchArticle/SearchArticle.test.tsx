import { render } from "@testing-library/react";
import SearchArticle from "./SearchArticle";
import { Provider } from "react-redux";
import { store } from "../../../Redux/store";

jest.mock("axios", () => ({
  data: { response: { docs: [] } },
}));

describe("Test SearchArticle", () => {
  it("Snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <SearchArticle />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
