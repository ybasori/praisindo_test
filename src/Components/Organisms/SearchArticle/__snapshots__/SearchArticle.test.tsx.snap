// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchArticle Snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <form
      class="container"
    >
      <input
        autocomplete="off"
        class="input"
        name="keyword"
        value=""
      />
      <button
        class="btn"
      >
        <i
          class="fas fa-search"
        />
      </button>
    </form>
    <div
      class="artilce-list"
    >
      <div
        class="container"
      >
        <div
          class="loading"
        >
          <i
            class="fas fa-circle-notch fa-spin"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
